src/lib.rs

```diff
@@ -681,6 +700,198 @@ impl<'a> Default for LocalExecutor<'a> {
     }
 }
 
+/// A leaked async [`Executor`].
+///
+/// Largely equivalent to calling `Box::leak(Box::new(executor))`, but spawning, running, and
+/// finishing tasks are optimized with the assumption that the executor will never be `Drop`'ed.
+/// A leaked executor may require signficantly less overhead in both single-threaded and
+/// mulitthreaded use cases.
+///
+/// As this type does not implement `Drop`, losing the handle to the executor or failing
+/// to consistently drive the executor with [`tick`] or [`run`] will cause the all spawned
+/// tasks to permanently leak. Any tasks at the time will not be cancelled.
+///
+/// Unlike [`Executor`], this type trivially implements both [`Clone`] and [`Copy`].
+///
+/// This type *cannot* be converted back to a `Executor`.
+#[derive(Copy, Clone)]
+#[cfg(feature = "leak")]
+pub struct LeakedExecutor {
+    state: &'static State,
+}
+
+#[cfg(feature = "leak")]
+// SAFETY: Executor stores no thread local state that can be accessed via other thread.
+unsafe impl Send for LeakedExecutor {}
+#[cfg(feature = "leak")]
+// SAFETY: Executor internally synchronizes all of it's operations internally.
+unsafe impl Sync for LeakedExecutor {}
+
+#[cfg(feature = "leak")]
+impl UnwindSafe for LeakedExecutor {}
+#[cfg(feature = "leak")]
+impl RefUnwindSafe for LeakedExecutor {}
+
+#[cfg(feature = "leak")]
+impl fmt::Debug for LeakedExecutor {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        debug_state(self.state, "Executor", f)
+    }
+}
+
+#[cfg(feature = "leak")]
+impl LeakedExecutor {
+    /// Spawns a task onto the executor.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use async_executor::Executor;
+    ///
+    /// let ex = Executor::new().leak();
+    ///
+    /// let task = ex.spawn(async {
+    ///     println!("Hello world");
+    /// });
+    /// ```
+    pub fn spawn<T: Send + 'static>(
+        &self,
+        future: impl Future<Output = T> + Send + 'static,
+    ) -> Task<T> {
+        let (runnable, task) = Builder::new()
+            .propagate_panic(true)
+            .spawn(|()| future, self.schedule());
+        runnable.schedule();
+        task
+    }
+
+    /// Spawns a non-`'static` task onto the executor.
+    ///
+    /// ## Safety
+    /// The caller must ensure that the returned task terminates
+    /// or is cancelled before the end of 'a.
+    pub unsafe fn spawn_scoped<'a, T: Send + 'static>(
+        &self,
+        future: impl Future<Output = T> + Send + 'a,
+    ) -> Task<T> {
```

Comment:
Would this kind of lifetime relaxation be useful on `Executor`/`LocalExecutor` as well?

Reply:
Perhaps, it would at the very minimum remove the need for `std::mem::transmute`s, which I would consider much more prone to unsoundness, to implement `ThreadPool::scope`-style APIs. Though it's pretty evidently not easy to properly use such an API, and proving the soundness even harder.

Reply:
that sounds interesting, could you elaborate?

Reply:
The primary API in question is [bevy_tasks::TaskPool::scope](https://docs.rs/bevy_tasks/0.13.2/bevy_tasks/struct.TaskPool.html#method.scope). The [actual implementation](https://github.com/bevyengine/bevy/blob/aa2ebbb43f388f2d300b5e72deac1012abbc9cd7/crates/bevy_tasks/src/task_pool.rs#L334) is visible there too though it's pretty tangled due to our need to explicitly schedule tasks onto a separate executor since `!Send` data must be accessed and dropped from the thread it was created from.